system/jlib/jstats.cpp

```diff
@@ -439,6 +439,16 @@ StringBuffer & formatStatistic(StringBuffer & out, unsigned __int64 value, Stati
     return formatStatistic(out, value, queryMeasure(kind));
 }
 
+constexpr static stat_type usTimestampThreshold = 100ULL * 365ULL * 24ULL * 3600'000000ULL; // 100 years in us
+// Support the gradual conversion of timestamps from us to ns.  If a timestamp is in the old format (before 2070)
+// then it is assumed to be in us, otherwise ns.
+stat_type normalizeTimestampToNs(stat_type value)
+{
+    if (value < usTimestampThreshold)
```

Comment:
I had concerns about this test (in general case, it seemed pretentious), now that I see this function is only called if SMeasureTimestampUs, it makes sense. But I suggest this function accept a StatisticMeasure as well and only operate on the appropriate measure types (unless we're trying to avoid unnecessary function calls and we can gurantee only Us timestamps will be provided).

Reply:
The name of the function makes it clear that it is a timestamp being normalized.  I could rename it to normalizeStat() and pass in the measure, but in at least one place that may make the calling code more complex.